spring-boot/proxy/src/main/java/eu/coatrack/proxy/metrics/MetricsCounterService.java

```diff
@@ -21,131 +21,95 @@
-        if (elements[0].equals("counter." + PREFIX)) {
-            log.debug("Metrics for transmission: " + inputMetric.getName());
+    private void incrementAndSetCount(Metric metricToTransmit) {
+        String counterId = createCounterIdFromMetric(metricToTransmit);
 
-            Metric outputMetric = new Metric();
-            outputMetric.setRequestMethod(elements[2]);
-            outputMetric.setCount(inputMetric.getValue().intValue());
-            outputMetric.setMetricsCounterSessionID(counterSessionID);
-            outputMetric.setType(MetricType.valueOf(elements[4]));
+        Counter counter = meterRegistry.find(counterId).counter();
+        if (counter == null){
+            counter = meterRegistry.counter(counterId);
+        }
+        counter.increment();
+        int currentCount = (int) counter.count();
+        metricToTransmit.setCount(currentCount);
+    }
 
-            // add a tmp api key object with just the key value, will later be mapped to entity on CoatRack admin side
-            ApiKey tmpApiKey = new ApiKey();
-            tmpApiKey.setKeyValue(elements[3]);
-            outputMetric.setApiKey(tmpApiKey);
+    private String createCounterIdFromMetric(Metric metric) {
+        StringJoiner stringJoiner = new StringJoiner(SEPARATOR);
+        stringJoiner.add(PREFIX)
+                .add(metric.getApiKey().getServiceApiName())
+                .add(metric.getRequestMethod())
+                .add(metric.getApiKey().getKeyValue())
+                .add(metric.getType().toString())
+                .add(String.valueOf(metric.getHttpResponseCode()))
+                .add(LocalDate.now().toString())
+                .add(metric.getPath());
+        return stringJoiner.toString();
+    }
 
-            String httpStatusString = elements[5];
-            Integer httpStatus = httpStatusString.equals("null") ? null : new Integer(httpStatusString);
-            outputMetric.setHttpResponseCode(httpStatus);
+    private Metric createMetricToTransmit(MetricsHolder mh) {
+        Metric metricToTransmit = new Metric();
 
-            String dateString = elements[6];
-            log.debug("date string is: {}", dateString);
-            LocalDate dateOfApiUsage = LocalDate.parse(dateString);
-            log.debug("date of api usage parsed: {}", dateOfApiUsage);
-            outputMetric.setDateOfApiCall(Date.valueOf(dateOfApiUsage));
+        ServiceApi serviceApi = new ServiceApi();
+        serviceApi.setName(mh.getServiceApiName().toString());
+        ApiKey apiKey = new ApiKey();
+        apiKey.setKeyValue(mh.getApiKeyValue());
+        apiKey.setServiceApi(serviceApi);
+        metricToTransmit.setApiKey(apiKey);
 
-            outputMetric.setPath(elements[7]);
+        metricToTransmit.setRequestMethod(mh.getRequestMethod());
+        metricToTransmit.setType(mh.getMetricType());
+        metricToTransmit.setHttpResponseCode(mh.getHttpResponseCode());
+        metricToTransmit.setDateOfApiCall(new Date());
```

Comment:
To be compatible with the previous version, I would suggest to do the following here:
```suggestion
        metricToTransmit.setDateOfApiCall(Date.valueOf(LocalDate.now()));
```

Reply:
Done as suggested.